vms/administrator/tests/test_report.py

```diff
@@ -54,367 +50,397 @@ def login_admin(self):
         created_job = create_job_with_details(job)
 
-        # create shift
-        shift = ['2017-08-21', '09:00', '15:00', '10', created_job]
+        # Create Shift
+        shift = ['2050-05-24', '09:00', '15:00', '10', created_job]
         created_shift = create_shift_with_details(shift)
 
-        # shift is assigned to volunteer-one, but hours have not been logged
-        volunteer_shift = register_volunteer_for_shift_utility(
-            created_shift, volunteer)
+        # Shift is assigned to volunteer-one, but hours have not been logged
+        report_page.go_to_admin_report()
+        register_volunteer_for_shift_utility(created_shift, volunteer)
+        report_page.get_page(self.live_server_url, PageUrls.administrator_report_page)
+        # Check admin report with null fields, should not return the above shift
+        # Using multiple tries so to cater to late loading of page.
+        for _ in range(3):
+            try:
+                report_page.fill_report_form(['', '', '', '', ''])
+                break
+            except NoSuchElementException:
+                pass
+        self.assertEqual(report_page.get_alert_box_text(), report_page.no_results_message)
+
+    @staticmethod
+    def register_dataset(parameters):
+        # Register dataset
+        # Register dataset
+        volunteer = create_volunteer_with_details_dynamic_password(parameters['volunteer'])
+        volunteer.organization = parameters['org']
+        volunteer.save()
 
-        report_page = self.report_page
-        # check admin report with null fields, should not return the above shift
-        report_page.fill_report_form(['', '', '', '', ''])
-        self.assertEqual(report_page.get_alert_box_text(),
-                         report_page.no_results_message)
+        # Create Event
+        event = parameters['event']
+        created_event = create_event_with_details(event)
 
+        # Create Job
+        job = parameters['job'] + [created_event]
+        created_job = create_job_with_details(job)
+
+        # Create Shift
+        shift = parameters['shift'] + [created_job]
+        created_shift = create_shift_with_details(shift)
 
-# Failing test case which has been documented - bug #327
-# Test commented out to prevent travis build failure
+        # Create VolunteerShift
+        log_hours_with_details(volunteer, created_shift, parameters['vshift'][0], parameters['vshift'][0])
 
-    def test_check_intersection_of_fields(self):
+    def create_dataset(self):
+        org_one = create_organization_with_details('org-one')
+        org_two = create_organization_with_details('org-two')
+        org_three = create_organization_with_details('org-three')
+        org_four = create_organization_with_details('org-four')
+        parameters = {'org': org_one,
+                      'volunteer': [
+                          'uname1',
+                          'uname1',
```

Comment:
and these kind of stuff should go to some other file. This makes this test a mile long. Please no.

Reply:
can you store these maybe in some JSON files and load it from the test and pass to the driver to populate ?

Reply:
Makes sense. :+1:

Reply:
Also just to point it out. The test using this JSON in commented due to incorrect results possible issue which wasn't fixed has been documented above it.

Reply:
Done.